### PR TITLE
 Views

### PR DESCRIPTION
**Pull request by massazoum**
Hi Team ,
In this project ,
- I Implemented the design from the sneak peek wireframes

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+<ul>
+  <% @users.each do |user| %>
+    <li class="user-list">
+      <%= user.name %>
+      <% if user.post_counter == 1 %>
+        (1 post)
+      <% else %>
+  <strong  class="Number_of_posts" >Number of posts : <%=user.posts.count %></strong>     
+      <% end %>
+    </li>
```
Comment (mikethreels) [inline app/views/users/index.html.erb]: - [ ] in the user index page you're supposed to show the user photo as well.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,21 @@
...
+</div>
+
+<h3>Posts:</h3>
+<ul>
+  <% @user.posts.each do |post| %>
+    <li class="post-list"><strong><%= post.title %></strong> <p><%= post.text %></p>   
+    <p class=detail_post> <strong  class="Number_of_comment"> Comments: <%=post.comments_counter %></strong> 
+      <strong  class="Number_of_like"> Likes : <%=post.likes_counter %></strong>  <p>
+    </li>   
+  <% end %>
```
Comment (mikethreels) [inline app/views/users/show.html.erb]: - [ ] According to the requirements you're not supposed to show all posts here only the recent posts, you can use the method created in the model for that

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,21 @@
...
+</ul>
```
Comment (mikethreels) [inline app/views/users/show.html.erb]: - [ ] According to the requirements there should be a button on the user show page redirecting to the user post path.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,2 +1,38 @@
...
+      <h3>Comments:</h3>
+      <% if post.comments.any? %>
+        <ul>
+          <% post.comments.each do |comment| %>
+            <li><%= comment.text %></li>
+          <% end %>
+        </ul>
+      <% else %>
+        <p>No comments for this post.</p>
+      <% end %>
```
Comment (mikethreels) [inline app/views/posts/index.html.erb]: - [ ] According to the requirements you're supposed to use the recent comments instead of all comments.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Massa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job creating the views

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -1,2 +1,26 @@
...
+<ul>
+  <% @users.each do |user| %>
+  <div class ="userphotoName">
+    <div class="photo"> 
+      <% if user.photo.present? %>
+        <%= image_tag user.photo, class: "user-photo" %>
+      <% else %>
+        <%= image_tag "default_avatar.png", alt: "Default Avatar", class: "user-photo" %>
+      <% end %>
+    </div>
```
Comment (franclobo) [inline app/views/users/index.html.erb:15]: - [ ] Well done implementing the required changes but we need to match the wireframe for a better UI, please 🙏🏽 adjust the photo size 👇🏽

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/users/index.html.erb [hunk 6]
```diff
@@ -1,2 +1,26 @@
...
+    <li class="user-list">
+      <span class="user-name"><%= user.name %></span>
```
Comment (franclobo) [inline app/views/users/index.html.erb:17]: - **[OPTIONAL]** I suggest using links for the names to be able to redirect to the user#show page.

File: app/views/users/show.html.erb
Comment (franclobo) [inline app/views/users/show.html.erb:1]: - **[OPTIONAL]** It is a project requirement to math the wireframe, kindly add the user's photo. 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -1 +1,27 @@
...
+<div class="bio">
+  <h1>Bio:</h1>
+  <%= @user.bio %>
+</div>
+
+<h3>Recent Posts:</h3>
+<ul>
+  <% @user.recent_posts.each do |post| %>
+    <li class="post-list">
+      <strong><%= post.title %></strong>
```
Comment (franclobo) [inline app/views/users/show.html.erb:36]: - [ ] As required by the last code reviewer 👇🏽 

>  According to the requirements there should be a button on the user show page redirecting to the user post path.

I recommend to use links on the post titles to redirect to the post#show page.

File: app/views/posts/index.html.erb
Comment (franclobo) [inline app/views/posts/index.html.erb:1]: - **[OPTIONAL]**  It is a project requirement to match the wireframe, please 🙏🏽 add the user's photo 👇🏽 

| Wireframe | Your design |
|---|---|
| image |image |

- **[OPTIONAL]** I suggest to use links on the post's name to redirect to the post#show page.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @massazoum, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @massazoum , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**